`std` feature of `rand_core` changes API

Specifically if you are calling `rand_core::Error::with_cause` with something that doesn't implement `std::error::Error` (because you don't directly rely on `std`) then another dependency activates the `rand_core/std` feature you will get a compilation error.

Simplest way to see this is to use `rand` as the library that is built on `rand_core` without the `std` feature, then activate the `std` feature in an empty library that depends on both:

```toml
[dependencies]
rand = { version = "0.6.5", default-features = false }
rand_core = { version = "0.4.0", default-features = false, features = ["std"] }
```

This will fail to build with the error:

```
error[E0277]: the trait bound `error::TimerError: std::error::Error` is not satisfied
  --> /home/wim/.cargo/registry/src/github.com-1ecc6299db9ec823/rand_jitter-0.1.3/src/error.rs:62:9
   |
62 |         Error::with_cause(ErrorKind::Unavailable,
   |         ^^^^^^^^^^^^^^^^^ the trait `std::error::Error` is not implemented for `error::TimerError`
   |
   = note: required because of the requirements on the impl of `core::convert::From<error::TimerError>` for `alloc::boxed::Box<(dyn std::error::Error + core::marker::Send + core::marker::Sync + 'static)>`
   = note: required because of the requirements on the impl of `core::convert::Into<alloc::boxed::Box<(dyn std::error::Error + core::marker::Send + core::marker::Sync + 'static)>>` for `error::TimerError`
   = note: required by `rand_core::Error::with_cause`
```

--- a/rand_jitter/CHANGELOG.md
+++ b/rand_jitter/CHANGELOG.md
@@ -4,6 +4,9 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.1.4] - 2019-05-02
+- Change error conversion code to partially fix #738
+
 ## [0.1.3] - 2019-02-05
 - Use libc in `no_std` mode to fix #723
 

--- a/rand_jitter/Cargo.toml
+++ b/rand_jitter/Cargo.toml
@@ -1,8 +1,8 @@
 [package]
 name = "rand_jitter"
-version = "0.1.3"
+version = "0.1.4"
 authors = ["The Rand Project Developers"]
-license = "MIT/Apache-2.0"
+license = "MIT OR Apache-2.0"
 readme = "README.md"
 repository = "https://github.com/rust-random/rand"
 documentation = "https://docs.rs/rand_jitter"

--- a/rand_jitter/src/error.rs
+++ b/rand_jitter/src/error.rs
@@ -59,8 +59,12 @@ impl From<TimerError> for Error {
     fn from(err: TimerError) -> Error {
         // Timer check is already quite permissive of failures so we don't
         // expect false-positive failures, i.e. any error is irrecoverable.
-        Error::with_cause(ErrorKind::Unavailable,
-                              "timer jitter failed basic quality tests", err)
+        #[cfg(feature = "std")] {
+            Error::with_cause(ErrorKind::Unavailable, "timer jitter failed basic quality tests", err)
+        }
+        #[cfg(not(feature = "std"))] {
+            Error::new(ErrorKind::Unavailable, "timer jitter failed basic quality tests")
+        }
     }
 }
 

